sources/api/migration/migrations/v1.14.1/aws-partition-setting/src/main.rs

```diff
@@ -0,0 +1,22 @@
+use migration_helpers::common_migrations::{AddSettingsMigration, NoOpMigration};
+use migration_helpers::{migrate, Result};
+use std::process;
+
+/// We added a new setting for AWS partition.
+fn run() -> Result<()> {
+    if cfg!(variant_platform = "aws") {
+        migrate(AddSettingsMigration(&["settings.aws.partition"]))
+    } else {
+        migrate(NoOpMigration)
+    }
+}
```

Comment:
This is not correct since `settings.aws.partition` will exist in all variants since `AwsSettings` is not conditionally compiled.
```suggestion
/// We added a new setting for AWS partition.
fn run() -> Result<()> {
    migrate(AddSettingsMigration(&["settings.aws.partition"]))
}
```